### ISSUE TITLE
Long Range AAM mode

### ISSUE THREAD

**Issue by Isenderen** [PapaJoesSoup/BDArmory#323]
##### ISSUE TYPE
 - Feature Idea (Something new you want to see)
 - Enhancement  (Expand on existing features)
(This is kind of a mix of a new feature and an enhancement to the current missile guidance modes feature)

##### KSP and BDA Version
KSP 1.3

Most likely an new BDA release

##### OS / ENVIRONMENT
Windows 10

##### SUMMARY
Currently the BDA missile guidance is good for what has been possible in previous versions of BD armory. That is shots within 20-30km from the target. However with physics range extender and the new radar range increase this brings the possibility of missile shots at closer to 90-100km. (i do know these shots were possible before the radar increase with legacy targeting, but that is with missiles that retain ~1g of force on them throughout the flight which is unrealistic except for missiles with ramjets like the meteor) So, i personally feel like it is time for an upgrade of the missile guidance from the simple just leading the target, to something more realistic where the missile will loft itself and gain altitude at the beginning of the flight in order to gain potential energy and glide down towards the target thereby increasing range drastically. However this should probably be a new guidance mode like LRAAM (Long Range Air to Air Missile) alongside the current AAM guidance mode. This need of a new missile guidance mode is especially apparent with the Procedural missiles and in non FAR ksp due to kerbin's increased thickness of air amplifying drag on the missiles, which in turn increases the speed of the P missiles losing energy once the motor has burned out.

So in short, please see if you can come up with a new AAM mode that incorporates lofting for longer ranged missile shots.

##### STEPS TO PRODUCE
1. program a guidance mode where the missile gains altitude to gain potential energy before slowly descending and tracking to the target (tbh i have no idea how you would do this, so im not going to dig myself into a hole trying to explain the step to doing this.)

**Comment by jrodrigv** [comment]
I like it. Although , I think it would not be very usable with stock KSP &  BDArmory.

But for RSS & RBDA...I think it might be interesting to see. 

Regarding the guidance.I'm thinking on a mix between the new ballistic guidance ( that allows to hit targets at > 500 km) using GPS and once we are close to the GPS target then activate the current AAM terminal guidance.

**Comment by Isenderen** [comment]
Even in stock KSP it would be useful, but not to the same degree as with FAR or RSS. It would need to be a flatter trajectory than it would in FAR or RSS though because it would need more speed.

Using the ballistic guidance would be a good idea too, but idk how well the gps would work. It would require more work, but wouldnt it be better to take the ballistic flight profile and have it guide towards the radar target?

**Comment by goduranus** [comment]
See the RBS missile, that is a GPS cruiser/Terminal active radar missile. A missile with GPS ballistic/Terminal active radar might approximate the behavior, although currently there is no mid-course correction ability for ballistic type missiles in BDA I think. But, according to wikipedia, some AMRAAMs also had no mid course correction ability during the loft flight, so at least some early type AMRAAMs could be approximated with this method.

**Comment by Isenderen** [comment]
That is true that early A model AMRAAMs didnt have mid course correction. However the feature was taken out pretty early on due to how often the missile failed to find its target once it hit the active guidance point, and this is why i said "idk how well it would work." When i said that i wasnt meaning i dont know how well the feature would work in game, i was meaning i dont know if people would think that it was worth the time to code and add in the mod. It would probably just be better to use a modified version of the ballistic flight profile and have the mid-course-correction only apply to guidance on the x axis during the loft phase.
